ceruleo/transformation/features/extraction.py

```diff
@@ -185,18 +210,14 @@ def __init__(self, *, feature: Optional[str] = None, name: Optional[str] = None)
         self.categories = set()
         self.encoder = None
 
-    def partial_fit(self, X: pd.DataFrame, y=None):
+    def partial_fit(self, X: pd.DataFrame, y=None) -> SimpleEncodingCategorical:
```

Comment:
It is not possible to type a method that returns a type of the class that contains it. 
We can put it as string or use Self.  Since python 3.11 is in typing but since we are supporting python 3.8. We need a workaround. There is a Self in the typing_extension module but i don't know the compatibility with the python version we support. 
https://realpython.com/python-type-self/

Reply:
Actually I inserted the type returned by a fit or partial_fit function only for these ones in which you saw the errors. On all the other I didn't write anything. In the documentation I just inserted the input parameters because I didn't think it was essential to say that they return an object of the class (like a SimpleEncodingCategorical object like in this case). But if you think we should include also this kind of information I can try to add it using the solution with Self